minidump-processor/src/op_analysis.rs

```diff
@@ -0,0 +1,733 @@
+                    let index = get_reg(reg)?;
+                    let scale = scale.unwrap_or(1);
+                    index.wrapping_mul(scale.into())
+                }
+                None => 0,
+            };
+
+            let disp = disp.unwrap_or(0) as i64 as u64;
+
+            Ok(base.wrapping_add(scaled_index).wrapping_add(disp))
+        };
+
+        let decoder = InstDecoder::default();
+        let decoded_instruction = decoder.decode_slice(bytes).map_err(|error| match error {
+            DecodeError::ExhaustedInput => OpAnalysisError::InstructionTruncated,
+            e => OpAnalysisError::DecodeFailed(e.into()),
+        })?;
+
+        let mut memory_accesses = Vec::new();
+
+        // Shortcut -- If the instruction doesn't access memory, just return an empty list
+        let mem_size = match decoded_instruction.mem_size() {
+            Some(access) => access.bytes_size(),
+            None => return Ok(memory_accesses),
+        };
+
+        for idx in 0..decoded_instruction.operand_count() {
+            let operand = decoded_instruction.operand(idx);
+
+            let maybe_address = match operand {
+                Operand::DisplacementU32(disp) => Some(disp.into()),
+                Operand::DisplacementU64(disp) => Some(disp),
+                Operand::RegDeref(base) => Some(calculate_address(Some(base), None, None, None)?),
+                Operand::RegDisp(base, disp) => {
+                    Some(calculate_address(Some(base), None, None, Some(disp))?)
+                }
+                Operand::RegScale(index, scale) => {
+                    Some(calculate_address(None, Some(index), Some(scale), None)?)
+                }
+                Operand::RegIndexBase(base, index) => {
+                    Some(calculate_address(Some(base), Some(index), None, None)?)
+                }
+                Operand::RegIndexBaseDisp(base, index, disp) => Some(calculate_address(
+                    Some(base),
+                    Some(index),
+                    None,
+                    Some(disp),
+                )?),
+                Operand::RegScaleDisp(index, scale, disp) => Some(calculate_address(
+                    None,
+                    Some(index),
+                    Some(scale),
+                    Some(disp),
+                )?),
+                Operand::RegIndexBaseScale(base, index, scale) => Some(calculate_address(
+                    Some(base),
+                    Some(index),
+                    Some(scale),
+                    None,
+                )?),
+                Operand::RegIndexBaseScaleDisp(base, index, scale, disp) => Some(
+                    calculate_address(Some(base), Some(index), Some(scale), Some(disp))?,
+                ),
+                _ => None,
+            };
+
+            if let Some(address) = maybe_address {
+                memory_accesses.push(MemoryAccess {
+                    address,
+                    size: mem_size,
```

Comment:
weird, all the different memory accesses share the same size?

Reply:
Operand size is encoded per-instruction on x86 and that should apply to all instructions (and in particular vector instructions can only have one memory operand IIRC)